From cb432fc77ae0f4c5da72a38ba6216d7243006dbe Mon Sep 17 00:00:00 2001
From: Pavel Zakharov <pavel.zakharov@delphix.com>
Date: Tue, 19 Jun 2018 13:14:27 -0400
Subject: [PATCH] 
 0026-Don-t-use-krb5-config-to-check-for-GSSAPI-on-Illumos.patch

---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index bad8657..5470ea1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4339,6 +4339,11 @@ AC_ARG_WITH([kerberos5],
 		AC_PATH_TOOL([KRB5CONF], [krb5-config],
 			     [$KRB5ROOT/bin/krb5-config],
 			     [$KRB5ROOT/bin:$PATH])
+		# Illumos has GSS but krb5-config doesn't talk about it
+		# Fall back to regular probing for libs
+		if uname -rs | grep 'SunOS 5.11' ; then
+			KRB5CONF=nonexistent
+		fi
 		if test -x $KRB5CONF ; then
 			K5CFLAGS="`$KRB5CONF --cflags`"
 			K5LIBS="`$KRB5CONF --libs`"
@@ -4380,7 +4385,7 @@ AC_ARG_WITH([kerberos5],
 					 AC_CHECK_LIB([des], [des_cbc_encrypt],
 					   [K5LIBS="$K5LIBS -ldes"])
 				       ], [ AC_MSG_RESULT([no])
-					 K5LIBS="-lkrb5 -lk5crypto -lcom_err"
+					 K5LIBS="-lkrb5"
 			])
 			AC_SEARCH_LIBS([dn_expand], [resolv])
 
-- 
2.14.3 (Apple Git-98)

